demos/keyboard_nav/index.html

```diff
@@ -251,7 +297,268 @@ <h1><a href="https://developers.google.com/blockly/">Blockly</a> &gt;
       if (oldCurNode) {
         Blockly.getMainWorkspace().getCursor().setLocation(oldCurNode);
       }
+      document.activeElement.blur();
+    }
+
+    // Start key mapping demo functions
+
+    /**
+     * Save the current key map in session storage.
+     * @package
+     */
+    function saveKeyMap() {
+      var currentMap = Blockly.user.keyMap.getKeyMap();
+      if (sessionStorage) {
+        sessionStorage.setItem('keyMap', JSON.stringify(currentMap));
+      }
+    }
+
+    /**
+     * Set the key map to the map from session storage.
+     * @package
+     */
+    function restoreKeyMap() {
+      var defaultMap = Blockly.user.keyMap.map_;
+      var stringifiedMap = sessionStorage.getItem('keyMap');
+      var restoredMap = {};
+      if (sessionStorage && stringifiedMap) {
+        var keyMap = JSON.parse(stringifiedMap);
+        for (var i = 0, key; key = Object.keys(keyMap)[i]; i++) {
```

Comment:
If you call Object.keys(keymap) outside the loop, it will be a bit more efficient, as right now I believe the list of keys is recreated each time the condition is evaluated.

Reply:
done